journals/apps/theming/models.py

```diff
@@ -18,7 +19,35 @@ class SiteBranding(BaseSetting):
 
     theme_name = models.CharField(max_length=128, blank=True, default='')
 
+    footer_links = JSONField(
+        verbose_name='Footer Links',
+        help_text="""JSON should be in the format: 
+                     [
+                       {
+                         'label_text': 'Link Name 1',
+                         'destination_link': 'link_url_1'
+                       }, 
+                       {
+                         'label_text': 'Link Name 2',
+                         'destination_link': 'link_url_2'
+                       },
+                    ]""",
+        null=False,
+        blank=False,
+        default=[
```

Comment:
JSON cannot start with a root element being an array. It must always be an object with any arrays inside of it. You can see an example of this [here](https://github.com/edx/journals/blob/master/journals/apps/journals/models.py#L81). 
```
video_course_ids = JSONField(
    verbose_name=_('Video Source Course IDs'),
    help_text=_('List of course IDs to pull videos from'),
    null=False,
    blank=False,
    default={'course_runs': []}
)
```

This was one of the nuisances I was hoping to avoid with the {"label":"url"} route. 

JSON also only supports double-quotes. Python probably is taking care of it on the backend (due to conversion to dict), but I would swap them out for consistency.

Reply:
so we need to have it be an array anyway b/c it won't remain ordered other wise
and i'll swap out the single quotes

Reply:
having it as an array didn't cause any problems for me - is there other issues I'm not aware of?

Reply:
Ignore all the stuff I said on arrays.. There was previously a security vulnerability if the top level element was an array, but that has been mitigated for years now. I had the "don't to this" in my head as a "can't do this". All good now though. I might go and fix the video_course_ids as well then.